apps/launchpad/app/.client/wallet-sync.tsx

```diff
@@ -0,0 +1,40 @@
+import { useEffect } from 'react'
+
+import { usePrivy } from '@privy-io/react-auth'
+import { useAccount, useDisconnect } from 'wagmi'
+
+export function WalletSync() {
+  const { address } = useAccount()
+  const { user, logout } = usePrivy()
+  const { disconnect } = useDisconnect()
+
+  useEffect(() => {
+    if (!user?.wallet?.address || !address) {
+      return
+    }
+
+    // If wallet addresses don't match, disconnect both
```

Comment:
We've had to solve this so many different ways. I hope that Privy improves this state handling at some point